### ISSUE TITLE
Create `category` tables

### ISSUE THREAD

**Issue by lache-melvin** [msupply-foundation/open-msupply#5330]
> Note: updated for category groups

For now:

```mermaid
erDiagram
    item ||--o{ category_item_join : id--item_id
    category ||--o{ category_item_join : id--category_id
    category_group ||--o{ category : id--category_group_id

    category_group {
        text id
        text name
        text root_id
    }

    category {
        text id
        text parent_id
        text description
        text name
        text category_group_id
    }
    category_item_join {
        text category_id
        text item_id
    }
    item {
        text id
    }
```

Categories exist in a hierarchy (e.g. Drug > Analgesic > NSAID) - if an item is an NSAID, CategoryGroup would be Drug, then it should have 2 `item_category_join` entries, one for each level, so item can easily be queried by category.

Future state: 

This design should allow for further extension in future, to include tags (allow us to query at a specific level of the hierarchy, across many branches easily) and categorise different domain types:

```mermaid
erDiagram
    category_group ||--o{ category : id--category_group_id
    item ||--o{ category_item_join : id--item_id
    category ||--o{ category_item_join : id--category_id
    category ||--o{ category_name_join : id--category_id
    name ||--o{ category_name_join : id--name_id
    category_group {
        text id
        text name
        text root_id
        domain_type domain
    }
    category {
        text id
        text parent_id
        text category_group_id
        text description
        text name
        domain_type domain
    }
    category_item_join {
        text category_id
        text item_id
    }
    
    category_item_join {
        text category_id
        text item_id
    }
    name {
        text id
    }
    item {
        text id
    }
    category_name_join {
        text category_id
        text name_id
    }
    domain_type {
        variant item
        variant name
    }
```

### Context

Andrei's original design: 

https://github.com/msupply-foundation/open-msupply/blob/c93ed5cc7ef3763aeda4cfcbd1de2cc49f651c7c/categories%20and%20properties.md

**Comment by lache-melvin** [comment]
Closed by #5343

**Comment by marthakeezy** [comment]
No specific testing steps in the linked issue, however tested the general functionality as a part of #5676 and moving to Done ✅
